sys/net/gnrc/network_layer/sixlowpan/iphc/gnrc_sixlowpan_iphc.c

```diff
@@ -1090,6 +1090,10 @@ static size_t _iphc_ipv6_encode(gnrc_pktsnip_t *pkt,
         if (src_ctx && !(src_ctx->flags_id & GNRC_SIXLOWPAN_CTX_FLAGS_COMP)) {
             src_ctx = NULL;
         }
+        /* bits not covered by context information must be zero */
+        if (src_ctx && ipv6_addr_match_prefix(&src_ctx->prefix, &ipv6_hdr->src) < 64) {
+            src_ctx = NULL;
```

Comment:
The comment says that the bits not covered by the context information must be zero, but if I understand the code correctly, the context is just not used if the prefix is less then 64. That's not the same as assuring the bits not covered by both prefix and IID are zero.

My naive approach would be: copy the address checked to a temporary address, set the bits of the prefix to 0, set the bits of the IID to 0 (in this case the IID is the generated 64-bit form in case of a 16-bit short address), check if the total address is zero, otherwise do not use the context. Since this requires knowledge if the source / destination is compressible at all, I would not put this here, but down below where the addresses are actually compressed.

Reply:
We _can_ do it like you propose as a workaround, but then the comment should reflect that, e.g.

```C
/* checking for 0-bits for short prefixes not supported at the moment */
```

Reply:
Unused bits in `ctx->prefix` are set to zero in `gnrc_sixlowpan_ctx_update()` by calling `ipv6_addr_set_unspecified(&(_ctxs[id].prefix))` - so when comparing the first 64 bits of the context to the address, the unused bits will be zero.

Reply:
:+1: Now, can we get rid somehow of that magic number?

Reply:
Do you have a suggestion in mind for a fitting name? `SIXLOWPAN_IPHC_PREFIX_LEN`?

Reply:
I was hoping, there is already a fitting name. I guess, `SIXLOWPAN_IPHC_PREFIX_LEN` is fine, as long as it is not defined in a global scope.